tests/disk/disk.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2021 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disk
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Disk has the attributes of a virtual disk which is emulated for integration
+// testing.
+type Disk struct {
+	// Size in bytes
+	Size int64
+	// The backing image name
+	// eg: /tmp/fake123
+	imageName string
+	// the disk name
+	// eg: /dev/loop9002
+	DiskPath string
+	// mount point if any
+	MountPoints []string
+}
+
+// NewDisk creates a Disk struct, with a specified size. Also the
+// random disk image name is generated. The actual image is generated only when
+// we try to attach the disk
+func NewDisk(size int64) Disk {
+	disk := Disk{
+		Size:      size,
+		imageName: generateDiskImageName(),
+		DiskPath:  "",
+	}
+	return disk
+}
+
+// Generates a random image name for the backing file.
+// the file name will be of the format fakeXXX, where X=[0-9]
+func generateDiskImageName() string {
+	rand.Seed(time.Now().UTC().UnixNano())
+	randomNumber := 100 + rand.Intn(899)
+	imageName := "fake" + strconv.Itoa(randomNumber)
+	return os.TempDir() + "/" + imageName
+}
+
+func (disk *Disk) createDiskImage() error {
+	f, err := os.Create(disk.imageName)
+	if err != nil {
+		return fmt.Errorf("error creating disk image. Error : %v", err)
+	}
+	err = f.Truncate(disk.Size)
+	if err != nil {
+		return fmt.Errorf("error truncating disk image. Error : %v", err)
+	}
+
+	return nil
+}
+
+// CreateLoopDevice creates a loopback device if the disk is not present
+func (disk *Disk) CreateLoopDevice() error {
+	if disk.DiskPath == "" {
+		var err error
+		if _, err = os.Stat(disk.imageName); err != nil && os.IsNotExist(err) {
+			err = disk.createDiskImage()
+			if err != nil {
+				return err
+			}
+		}
```

Comment:
Missing below else part else original error will be swallowed
```suggestion
		} else if err != nil {
		     return err
		}
```